### ISSUE TITLE
nats-async does not compile with latest rustls

### ISSUE THREAD

**Issue by turulix** [nats-io/nats.rs#1018]
### NATS version

0.29.0

### rustc version

1.73.0-nightly

### OS/Container environment

Ubuntu latest (What ever github runner uses)

### Steps or code to reproduce the issue

1. Try to compile it within a new project.

Seems to be related to https://github.com/rustls/rustls/commit/b3a0ce23f9c131d6dc2cf284496cf6c6846c274c where they changed the function signature with a patch release .-.

### Expected result

I'd expect it to compile.

### Actual result

```rust
error[E0[282](https://github.com/turulix/memphis-rust-community/actions/runs/5384718935/jobs/10046570200#step:6:283)]: type annotations needed
  --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-nats-0.29.0/src/tls.rs:69:20
   |
69 |         root_store.add_parsable_certificates(
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type for type parameter `C` declared on the method `add_parsable_certificates`

error: future cannot be sent between threads safely
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-nats-0.29.0/src/lib.rs:728:17
    |
728 |       task::spawn(async move {
    |  _________________^
729 | |         if connection.is_none() && options.retry_on_initial_connect {
730 | |             let (info, connection_ok) = connector.connect().await.unwrap();
731 | |             info_sender.send(info).ok();
...   |
742 | |         connection_handler.process(receiver).await
743 | |     });
    | |_____^ future created by async block is not `Send`
    |
note: opaque type is declared here
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-nats-0.29.0/src/tls.rs:65:63
    |
65  | pub(crate) async fn config_tls(options: &ConnectorOptions) -> io::Result<rustls::ClientConfig> {
    |                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: this item depends on auto traits of the hidden type, but may also be registering the hidden type. This is not supported right now. You can try moving the opaque type and the item that actually registers a hidden type into a new submodule
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-nats-0.29.0/src/lib.rs:671:14
    |
671 | pub async fn connect_with_options<A: ToServerAddrs>(
    |              ^^^^^^^^^^^^^^^^^^^^
note: future is not `Send` as it awaits another future which is not `Send`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-nats-0.29.0/src/connector.rs:298:17
    |
298 |                 tls::config_tls(&self.options)
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ await occurs here on type `impl futures::Future<Output = Result<ClientConfig, std::io::Error>>`, which is not `Send`
note: required by a bound in `tokio::spawn`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/task/spawn.rs:166:21
    |
164 |     pub fn spawn<T>(future: T) -> JoinHandle<T::Output>
    |            ----- required by a bound in this function
165 |     where
166 |         T: Future + Send + 'static,
    |                     ^^^^ required by this bound in `spawn`
```

**Comment by turulix** [comment]
Related to #1017

**Comment by paolobarbolini** [comment]
Related to https://github.com/rustls/rustls/issues/1352

**Comment by n1ghtmare** [comment]
The rustls team issued a hotfix and it should now be resolved with rustls `v0.21.5`. https://github.com/rustls/rustls/issues/1352

**Comment by Jarema** [comment]
rustls reverted the breaking change

https://github.com/rustls/rustls/releases/tag/v%2F0.21.5

Library builds fine again.

Closing.
